prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -108,21 +108,18 @@ class ReportFunction : Logging {
             PrincipalLevel.USER,
             workflowEngine
         )
-
-        // replace to make the senderName match with the ReportStream database (dashes instead of underscores)
-        // later, in the Okta Verifier,
-        // all the dashes in the sender.fullName are turned into underscores to match the Okta Organization naming scheme
-        val senderName = request.headers[CLIENT_PARAMETER]?.replace('_', '-')
-            ?: request.queryParameters.getOrDefault(CLIENT_PARAMETER, "").replace('_', '-')
-        // todo This code is redundant w/validateRequest. Remove from validateRequest once old endpoint is removed
-        if (senderName.isBlank())
+        val senderName = extractClientHeader(request)
+        if (senderName.isNullOrBlank())
             return HttpUtilities.bad(request, "Expected a '$CLIENT_PARAMETER' query parameter")
         val sender = workflowEngine.settings.findSender(senderName)
             ?: return HttpUtilities.bad(request, "'$CLIENT_PARAMETER:$senderName': unknown sender")
         if (authenticationStrategy is OktaAuthentication) {
-            // Okta Auth
+            // The oktaSenderName is the same as the sender.fullName, but with dashes instead of underscores.
+            // later, in the Okta Verifier,
+            // the dashes in the oktaSenderName are turned into underscores to match the Okta Organization naming scheme
+            val oktaSenderName = sender.fullName.replace('_', '-')
```

Comment:
This is the bug fix here - only the Okta code wants the whole dash/underscore replacement stuff.